spec/takeaway_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'takeaway'
+
+describe Takeaway do
+  let(:menu_klass) { double(:menu_klass, new: menu) }
+  let(:dishes) { {menuitem1: 1, menuitem2: 1.50 } }
+  let(:menu) { double(:menu, show: dishes, dishes: dishes)}
+
+  subject(:takeaway) { described_class.new(menu_klass: menu_klass) }
+
+  describe '#show_menu' do
+    it 'shows a list of menu items' do
+      expect(takeaway.show_menu).to eq dishes
+    end
+  end
+
+  describe '#order' do
+    context 'item not available' do
+      it 'raises an error' do
+        msg = 'This item is not on the menu'
+        expect {takeaway.order('Hamburger', 25)}.to raise_error msg
+      end
+    end
+    context 'first order' do
+      it 'stores name and quantity of dish in basket' do
+        takeaway.order(:menuitem1, 2)
+        expect(takeaway.basket).to include({menuitem1: 2})
+      end
+    end
+    context 'additional orders' do
+      it 'stores name and quantity of dish in basket' do
+        takeaway.order(:menuitem1, 2)
+        takeaway.order(:menuitem1, 2)
+        expect(takeaway.basket).to include({menuitem1: 4})
+      end
+    end
+  end
+
+  describe '#checkout' do
+    context 'when total given by user corresponds to calculated total' do
+      it 'sends a confirmation text message' do
+        takeaway.order :menuitem1, 2
+        expect(takeaway).to receive(:send_text).with("menuitem1 x2: £2", "Total price: £2")
```

Comment:
Metrics/LineLength: Line is too long. [91/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)